frontend/src/pages/upload/UploadPage.tsx

```diff
@@ -10,6 +14,55 @@ import {
 import { ExperimentalMeasurementDto } from "../../api/types";
 import "../../components/Spinner.scss";
 
+type CompoundMap = { [compoundId: string]: ExperimentalMeasurementDto[] };
+
+function groupCompounds(mes: ExperimentalMeasurementDto[]): CompoundMap {
+  return mes.reduce((rv: CompoundMap, x) => {
+    // eslint-disable-next-line no-param-reassign
+    (rv[x.compoundId] = rv[x.compoundId] || []).push(x);
```

Comment:
Since I didn't write in TS before, this syntax looks quite weird to me. Can you explain what's going on here please?

Reply:
there goes into action implicit conversions, like in C++: if `rv[x.compoundId]` exists then it will be returned and and put an key `x.compoundId` (so nothing will change), and it will be returned from braces and new element pushed in this list (as list is reference this will take effect in map).

otherwise, if `rv[x.compoundId]` doesn't exist, so `undefined` || with `[]` will give `[]`, which will be assigned to a key `x.compoundId`, and new element pushed in it.

I will rewrite this piece of code perhaps